lcm/lcm_file.c

```diff
@@ -355,22 +355,20 @@ lcm_logprov_publish (lcm_logprov_t *lcm, const char *channel, const void *data,
     return 0;
 }
 
-static lcm_provider_vtable_t logprov_vtable;
+static lcm_provider_vtable_t logprov_vtable = {
+    .create      = lcm_logprov_create,
+    .destroy     = lcm_logprov_destroy,
+    .subscribe   = NULL,
+    .unsubscribe = NULL,
+    .publish     = lcm_logprov_publish,
+    .handle      = lcm_logprov_handle,
+    .get_fileno  = lcm_logprov_get_fileno
+};
 static lcm_provider_info_t logprov_info;
 
-
 void
 lcm_logprov_provider_init (GPtrArray * providers)
 {
-// Microsoft VS compiler issues. Can't do this statically
```

Comment:
Hmm... this is not reassuring. Can someone verify if this works with at least VS2015? (Alternatively, I suppose we could `#ifdef` this mess on platform until Microsoft fixes their compiler.)